src/main/java/resources/PairResource.java

```diff
@@ -0,0 +1,253 @@
+package resources;
+
+import com.google.inject.Inject;
+import dao.SlotsDao;
+import domain.Pair;
+import domain.Persons;
+import domain.Slots;
+import io.dropwizard.hibernate.UnitOfWork;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+@Path("/pairs")
+@Produces(MediaType.APPLICATION_JSON)
+public class PairResource {
+
+    private SlotsDao slotsDao;
+
+    @Inject
+    public PairResource(SlotsDao slotsDao) {
+        this.slotsDao = slotsDao;
+    }
+
+    @GET
+    @UnitOfWork
+    public List<Persons> findPairs(@QueryParam("startDate") String startDate,
+                                   @QueryParam("endDate") String endDate,
+                                   @QueryParam("isDev") Boolean isDev,
+                                   @QueryParam("isTest") Boolean isTest,
+                                   @QueryParam("isOps") Boolean isOps) {
+
+        List<Slots> slots = slotsDao.findBetweenPerJobProfile(startDate, endDate, isDev, isTest, isOps);
+        List<Persons> persons = findPersonsInSlots(slots);
+        List<Slots> filteredSlots = findTriplesInSlots(slots);
+        List<Pair> pairs = findPairsForAllPersons(filteredSlots, persons);
+
+        List<Persons> pp = mergePairsIntoPersons(pairs);
```

Comment:
This is meant for all those private methods. The resource should only call a service method and return the computed value.